packages/cli/src/commands/git/connect.ts

```diff
@@ -101,10 +121,194 @@ export default async function connect(
+  if (typeof connect === 'number') {
+    return connect;
+  }
+  output.log(
+    `Connected ${formatProvider(provider)} repository ${chalk.cyan(repoPath)}!`
+  );
+  return 0;
+}
+
+async function connectArgWithLocalGit(
+  client: Client,
+  output: Output,
+  org: Org,
+  project: Project,
+  confirm: boolean,
+  gitConfig: Dictionary<any>,
+  repoArg: string
+) {
+  const parsedUrlArg = parseRepoUrl(repoArg);
+  if (!parsedUrlArg) {
+    output.error(
+      `Failed to parse URL "${repoArg}". Please ensure the URL is valid.`
+    );
+    return 1;
+  }
+
+  const remoteUrls = pluckRemoteUrls(gitConfig);
+  if (remoteUrls) {
+    const shouldConnect = await promptConnectArg(
+      client,
+      output,
+      confirm,
+      repoArg,
+      remoteUrls
+    );
+    if (!shouldConnect) {
+      return 1;
+    }
+    if (shouldConnect) {
+      const { provider, org: parsedOrg, repo } = parsedUrlArg;
+      const repoPath = `${parsedOrg}/${repo}`;
+      output.log(`Connecting Git remote: ${link(repoArg)}`);
+      const connect = await checkExistsAndConnect(
+        client,
+        output,
+        confirm,
+        org,
+        project,
+        project.link,
+        provider,
+        repoPath,
+        parsedOrg,
+        repo
+      );
+      if (typeof connect === 'number') {
+        return connect;
+      }
+      output.log(
+        `Connected ${formatProvider(provider)} repository ${chalk.cyan(
+          repoPath
+        )}!`
+      );
+    }
+    return 0;
+  }
+  return await connectArg(client, output, confirm, org, project, repoArg);
+}
+
+async function promptConnectArg(
+  client: Client,
+  output: Output,
+  yes: boolean,
+  arg: string,
+  remoteUrls: Dictionary<string>
+) {
+  const multiple = Object.keys(remoteUrls).length > 1;
+  if (multiple) {
+    output.log(`Found multiple Git repositories in your local Git config:`);
+    printRemoteUrls(output, remoteUrls);
+  } else {
+    const url = Object.values(remoteUrls)[0];
+    const parsedUrl = parseRepoUrl(url);
+    if (!parsedUrl) {
+      output.error(
+        `Failed to parse URL "${url}". Please ensure the URL is valid.`
+      );
+      return false;
+    }
+    const parsedUrlArg = parseRepoUrl(arg);
+    if (!parsedUrlArg) {
+      output.error(
+        `Failed to parse URL "${arg}". Please ensure the URL is valid.`
+      );
+      return false;
+    }
+    if (JSON.stringify(parsedUrl) === JSON.stringify(parsedUrlArg)) {
+      return true;
+    }
+
+    output.log(
+      `Found a repository in your local Git Config: ${chalk.cyan(
+        Object.values(remoteUrls)[0]
+      )}`
+    );
+  }
 
+  let shouldConnect = yes;
+  if (!shouldConnect) {
+    shouldConnect = await confirm(
+      client,
+      `Do you still want to connect ${chalk.cyan(arg)}?`,
+      false
+    );
+    if (!shouldConnect) {
+      output.log(`Aborted. Repo not connected.`);
+    }
+  }
+  return shouldConnect;
+}
+
+async function checkExistsAndConnect(
+  client: Client,
```

Comment:
10 function arguments is a lot. Consider passing in a single argument which is an object that is destructured. It's too easy to confuse the arguments, especially if another argument is added or an argument is removed.

```javascript
interface GitRepoCheckParams {
  client: Client;
  confirm: boolean;
  gitProviderLink?: ProjectLinkData;
  org: Org;
  output: Output;
  parsedOrg: string;
  project: Project;
  provider: string;
  repo: string;
  repoPath: string;
}

async function checkExistsAndConnect({
  client,
  confirm,
  gitProviderLink
  org,
  output,
  parsedOrg,
  project,
  provider,
  repoPath
}: GitRepoCheckParams) {
  // snip
```

Reply:
Thanks, did a bit of refactoring.